src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory extends StreamFactory
```

Comment:
> why shouldnt the MultipartStreamFactory not create simple streams to?

Our **implementation** of `MultipartStreamFactory` should also implement `StreamFactory`. But the interfaces has nothing to do with each other and there is no (extra high) probability that if you use `MultipartStreamFactory` you will also be using a `StreamFactory`